dashboard/lib/api/v1/pd/workshop_score_summarizer.rb

```diff
@@ -151,15 +151,42 @@ def get_score_for_workshops(workshops:, include_free_responses:, facilitator_nam
   # Return a hash of facilitator name to how many responses they have
   # @param responses [Array(Hash)] List of responses in hash form
   # @return [Hash] Hash of names to response counts
-  def calculate_facilitator_name_frequencies(responses)
-    if responses.first['how_often_given_feedback_s'].is_a? Hash
+  def calculate_facilitator_name_frequencies(responses, workshops)
+    facilitator_names = responses.map {|x| x['who_facilitated_ss']}.flatten.uniq
+
+    if facilitator_names == [nil]
+      # No workshops have facilitator specific answers
+      return nil
+    elsif facilitator_names.include? nil
+      # This means that we have a hybrid of facilitator specific responses and non
+      # facilitator specific responses. So we assume that each response is for all
+      # facilitators and add to responses accordingly
+
+      # We need a map of workshop_ids to facilitators. Here, we use it to update the
+      # response count for non-facilitator specific responses
+      workshops_to_facilitators = workshops.joins(:facilitators).includes(:facilitators).map {|x| [x.id, x.facilitators.map(&:name)]}.to_h
+
+      facilitator_names = (facilitator_names + workshops_to_facilitators.values.flatten).uniq
+
+      facilitator_name_responses = Hash[facilitator_names.compact.map {|facilitator| [facilitator, 0]}]
+
+      responses.each do |response|
+        if response['who_facilitated_ss']
+          response['who_facilitated_ss'].each do |facilitator|
+            facilitator_name_responses[facilitator] += 1
+          end
+        else
+          workshops_to_facilitators[response['workshop_id_i']].each {|facilitator| facilitator_name_responses[facilitator] += 1}
+        end
+      end
+
+      return facilitator_name_responses, workshops_to_facilitators
+    else
```

Comment:
This seems unnecessarily complex to me. Why do we need the special cases? The above `elsif` block handles a mix of facilitator-specific and non-facilitator-specific questions, so can't that be the general (and only) case? Why do we need the `else` below? 

It also looks like we don't need the initial `if facilitator_names == [nil]` case, again this could be handled generally in one branch, though it could also be a useful short circuit.